aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -1220,6 +1226,18 @@ func resourceAviatrixSpokeGatewayCreate(d *schema.ResourceData, meta interface{}
 		}
 	}
 
+	enableSpokePreserveAsPath := d.Get("enable_preserve_as_path").(bool)
+	if enableBgp {
+		if enableSpokePreserveAsPath {
+			err := client.EnableSpokePreserveAsPath(gateway)
+			if err != nil {
+				return fmt.Errorf("could not enable spoke preserve as path: %v", err)
+			}
+		}
+	} else {
+		return fmt.Errorf("enable_preserve_as_path only work for the BGP spoke gateway")
```

Comment:
enable_preserve_as_path is optional. Why do we return error here for non-bpg spoke?